File: backend/src/Core/Dev4All.Domain/Entities/GitHubLog.cs
using Dev4All.Domain.Common;
using Dev4All.Domain.Exceptions;

namespace Dev4All.Domain.Entities;

public class GitHubLog : BaseEntity
{
    public Guid ProjectId { get; private set; }
    public string RepoUrl { get; private set; } = string.Empty;
    public string Branch { get; private set; } = "main";
    public string CommitHash { get; private set; } = string.Empty;
    public string CommitMessage { get; private set; } = string.Empty;
    public string AuthorName { get; private set; } = string.Empty;
    public DateTime PushedAt { get; private set; }

    public Project Project { get; set; } = null!;

    public static GitHubLog Create(
        Guid projectId, string repoUrl, string branch, string commitHash, string commitMessage, string authorName, DateTime pushedAt)
    {
        if (projectId == Guid.Empty)
            throw new BusinessRuleViolationException("Project ID cannot be empty.");

        return new GitHubLog
        {
            ProjectId = projectId,
            RepoUrl = repoUrl,
            Branch = branch,
            CommitHash = commitHash,
            CommitMessage = commitMessage,
            AuthorName = authorName,
            PushedAt = pushedAt
        };
    }
}

File: mobile/core/domain/src/main/kotlin/com/dev4all/mobile/core/domain/model/GitHubLog.kt
package com.dev4all.mobile.core.domain.model

/**
 * Domain model for a GitHub activity log entry (commit).
 * Field names match backend Dev4All.Domain.Entities.GitHubLog.
 */
data class GitHubLog(
    val id: String,
    val projectId: String,
    val repoUrl: String,
    val branch: String = "main",
    val commitHash: String,
    val commitMessage: String,
    val authorName: String,
    val pushedAt: String,
)
